### PR TITLE
Adding ROS on Azure with Linux VM

### PR DESCRIPTION
**Pull request by seanyen**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Adding `ROS on Azure with Linux VM`, which is a equivalent setup to `ros-vm-windows` but in Linux. 
*
*

### CODE REVIEW

**Comment by seanyen** [comment]
This is ready for review and merge. Thanks.

File: ros-vm-linux/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        "adminPasswordOrKey": {
+            "type": "securestring",
+            "metadata": {
+                "description": "SSH Key or password for the Virtual Machine. SSH key is recommended."
+            }
+        },
+        "virtualMachineSize": {
+            "type": "string",
+            "defaultValue": "Standard_D2_v2",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline ros-vm-linux/azuredeploy.json:45]: remove allowedValues

File: ros-vm-linux/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+        },
+        {
+            "type": "Microsoft.Network/publicIpAddresses",
+            "apiVersion": "2019-02-01",
+            "name": "[variables('publicIpAddressName')]",
+            "location": "[variables('location')]",
+            "properties": {
+                "publicIpAllocationMethod": "Dynamic",
+                "dnsSettings": {
+                    "domainNameLabel": "[variables('domainNameLabel')]"
```
Comment (bmoore-msft) [inline ros-vm-linux/azuredeploy.json:235]: this needs to be globally unique

File: ros-vm-linux/nestedtemplates/customScriptExtension.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "name": "[concat(parameters('vmName'), '/', variables('extensionNameLinux'))]",
+            "location": "[parameters('location')]",
+            "properties": {
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.0",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                    "fileUris": "[parameters('fileUris')]",
+                    "commandToExecute": "[parameters('commandToExecute')]"
```
Comment (bmoore-msft) [inline ros-vm-linux/nestedtemplates/customScriptExtension.json:46]: move this to protectedSettings since it may contain secrets (the PATs)

**Comment by seanyen** [comment]
Thanks for the review!

**Comment by seanyen** [comment]
@bmoore-msft A quick question, when will be the next sync to the https://azure.microsoft.com/en-us/resources/templates/? Just want to know when the new template will be showing up on the portal. :)

**Pull request merged**